### PR TITLE
feature/livestream

### PR DESCRIPTION
**Pull request by ephraimeg**
Issue #2 
- Authenticated user can create an appeal 
- Authenticated user can view list of appeals
- Appeal owner can set description of the appeal
- Appeal owner can edit the appeal
- Appeal owner can cancel an appeal
- Appeal owner and helper can retrieve appeal instance

### CODE REVIEW

File: livestream/views.py [hunk 1]
```diff
@@ -65,50 +65,22 @@ def retrieve(self, request, *args, **kwargs):
...
+            token = opentok.generate_token(appeal.session_id)
+            # check if token is created successfully
+            if token:
+                self.context = {
+                    'API_KEY': OPENTOK_API,
+                    'SESSION_ID': appeal.session_id,
+                    'TOKEN': token,
+                }
+                serializer = AppealSerializer(appeal, context={'token': token})
+                print(serializer.data)
```
Comment (jonathantumulak) [inline livestream/views.py]: @ephraimeg please remove this print statement

File: livestream/views.py [hunk 2]
```diff
@@ -65,50 +65,22 @@ def retrieve(self, request, *args, **kwargs):
...
-        print(token)
-        self.context = {
-            'API_KEY': OPENTOK_API,
-            'SESSION_ID': appeal.session_id,
-            'TOKEN': token,
-        }
-        print(self.context)
-
-        return self.render_to_response(self.context)
+        if user == appeal.owner or user == appeal.helper:
```
Comment (jonathantumulak) [inline livestream/views.py]: @ephraimeg can you do a fail first here? just directly do the return if user is not owner or helper.

Comment (jonathantumulak) [reply livestream/views.py]: oh wait nvm this is just for testing right?

File: livestream/views.py [hunk 3]
```diff
@@ -52,6 +56,17 @@ def list(self, request, *args, **kwargs):
         serializer = self.get_serializer(queryset, many=True)
         return Response(serializer.data)
 
+    def __serialize_token(self, appeal, user):
+        if (user == appeal.owner or user == appeal.helper):
```
Comment (jonathantumulak) [inline livestream/views.py]: @ephraimeg these conditions are hard to trace.

maybe you can do
```
if not owner and not helper:
    return
if not token:
    return
return

File: userprofile/serializers.py [hunk 4]
```diff
@@ -22,6 +22,7 @@ class Meta:
                   'online', 'openappeals', 'closedappeals', 'offers')
 
     def get_offers(self, obj):
+
         openappeals = AppealSerializer(obj.offers.exclude(is_active=False),
                                        many=True)
         closedappeals = AppealSerializer(
```
Comment (aldnav) [inline userprofile/serializers.py]: Perhaps a good suggestion would be a better semantic api for appeals

`api/appeals/{int:userid}/?status=[open|close]`

Just think about it again and do it on another PR if you consider that helpful.

You can lookup semantic, api, guidelines, best practices, etc.,

Comment (aldnav) [reply userprofile/serializers.py]: @ephraimeg @allenazamoras
